lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1751,11 +1765,27 @@ static void applyOverrideOptions(std::vector<std::string> &args,
 void SwiftASTContext::AddExtraClangArgs(
     const std::vector<std::string> &ExtraArgs, StringRef overrideOpts) {
   swift::ClangImporterOptions &importer_options = GetClangImporterOptions();
-  AddExtraClangArgs(ExtraArgs, importer_options.ExtraArgs);
+
+  // Detect cc1 flags.  When DirectClangCC1ModuleBuild is on then the
+  // clang arguments in the serialized invocation are clang cc1 flags,
+  // which are very specific to one compiler version and cannot
+  // be merged with driver options.
+  bool fresh_invocation = importer_options.ExtraArgs.empty();
+  if (fresh_invocation && !ExtraArgs.empty() && ExtraArgs.front() == "-cc1")
+    importer_options.DirectClangCC1ModuleBuild = true;
```

Comment:
should `DirectClangCC1ModuleBuild` always be set to try when `ExtraArgs.front() == "-cc1"`?

in other words, should the mixed mode treat extra args as frontend flags when `-cc1` is first?

Reply:
I think such a case is hopeless anyway. I'm planning to clean this up on the rebranch branch, where I'm going to rip out support for non-precise compiler invocations, which should allow us to simplify all this code quite a bit.